spring-cloud-kubernetes-istio/src/main/java/org/springframework/cloud/kubernetes/istio/profile/IstioProfileApplicationListener.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *     Copyright (C) 2016 to the original authors.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *             http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.istio.profile;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.boot.context.event.ApplicationEnvironmentPreparedEvent;
+import org.springframework.cloud.kubernetes.istio.utils.MeshUtils;
+import org.springframework.context.ApplicationListener;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
+
+public class IstioProfileApplicationListener implements ApplicationListener<ApplicationEnvironmentPreparedEvent>, Ordered {
```

Comment:
One place we've not been shy about going over the network is with `BootstrapConfiguration` (see config client, consul, and zookeeper). Let's move the contacting of the local envoy to a `BootstrapConfiguration`. You should be able to use `@ConfigurationProperties` to parameterize host/port etc... Have it add a boolean property to the environment (`spring.cloud.istio.enabled=true` or something), then code like this later can just check the value of that property and activate the profile.